src/prefect/client/client.py

```diff
@@ -914,16 +914,22 @@ def get_default_tenant_slug(self, as_user: bool = True) -> str:
             slug = res.get("data").tenant[0].slug
         return slug
 
-    def create_project(self, project_name: str, project_description: str = None) -> str:
+    def create_project(
+        self,
+        project_name: str,
+        project_description: str = None,
+        skip_if_exists: bool = False,
+    ) -> str:
         """
         Create a new Project
 
         Args:
             - project_name (str): the project that should contain this flow
             - project_description (str, optional): the project description
+            - skip_if_exists (bool, optional): skip creating the project if it already exists
 
         Returns:
-            - str: the ID of the newly-created project
+            - str: the ID of the newly-created project or an empty string if it already exists
```

Comment:
Let's change this to returning None instead of an empty string. Will need to update the type of the function from 

```
-> str
```
to
```
-> Optional[str]
```

Reply:
I think we should return the id regardless - if it already exists we could lookup and return the existing ID.

Thinking about how this might be used, I think most users probably want this function to ensure a project exists, and are less likely to need to branch based on if the project exists already or not. Being consistent in return value (always returning the project id) makes it easier for the user to write their logic without having to check for a `None` result.

We might even make the idempotency the default (or only behavior)? I can't think of a reason I'd want to have `create_project` error if the project already existed.

Reply:
I have changed it to always return the id, so the return type hint remains `str`.

I like the idea of making idempotency the only behaviour. Shall I go ahead with that?

Reply:
Also, I have no idea why the code style checks are failing: both `black .` and `mypy src` are fine for me.

Reply:
Ah okay, I checked the circleci config and figured out that `flake8 src` was being run, and turns out I needed to fix something with that. I think `dev-requirements.txt` might need to include `flake8` and the contributing docs might need to be updated to mention running `flake8 src`.